### PR TITLE
Update responses for twilio voice channel

### PR DESCRIPTION
**Pull request by hsm207**
Trying out the twilio voice connector.

Summary of conversation related changes:

Channel specific responses
* Rewrote utterances with emojis, abbreviations, buttons, asking to type commands e.g. help

### CODE REVIEW

File: domain.yml
Comment (hsm207) [inline domain.yml:220]: Does having buttons in a voice channel make sense?

It does not feel natural to me especially if the choice is between "Yes" or "No".

When the bot finished asking "Would you like to transfer ... ?", I can't answer immediately because I need to wait for it to finish uttering each button's text representation i.e. "Yes" and "No, cancel the transactions".

File: domain.yml
Comment (hsm207) [inline domain.yml:130]: Emojis don't work with voice

File: domain.yml
Comment (hsm207) [inline domain.yml:143]: Abbreviations don't work with voice

File: domain.yml
Comment (hsm207) [inline domain.yml:145]: Abbreviations don't work with voice

File: domain.yml
Comment (hsm207) [inline domain.yml:132]: Emojis don't work with voice

File: domain.yml
Comment (hsm207) [inline domain.yml:221]: Does having buttons in a voice channel make sense?

It does not feel natural to me especially if the choice is between "Yes" or "No".

When the bot finished asking "Would you like to transfer ... ?", I can't answer immediately because I need to wait for it to finish uttering each button's text representation i.e. "Yes" and "No, cancel the transactions".

**Review by ArjaanBuijk — CHANGES_REQUESTED** [review]
@hsm207 ,
This looks really cool. I did not get it to work fully, but I am getting close I think.

I provided comments where I think additional clarifications will be helpful.

File: README-voice.md
Comment (ArjaanBuijk) [reply README-voice.md:31]: Tried after retraining, but still get the same error.

File: README-voice.md
Comment (hsm207) [reply README-voice.md:1]: This is a great idea!

Since we can't version control wiki pages, I'll transfer the content of README-voice.md to the wiki after we've finalized it so that it's easier to discuss changes.

File: README-voice.md
Comment (hsm207) [reply README-voice.md]: Does this work?

I provided a link to the download page so readers that follow the official instructions to install on their respective platform.

File: README-voice.md
Comment (hsm207) [reply README-voice.md]: > How do I test that ngrok is working? Maybe add a section:
> 
> ```shell
> # test ngrok is working
> ./ngrok --version
> ```

done

Comment (hsm207) [reply README-voice.md]: > After these steps I have `ngrok` and `ngrok-...*.zip` in the project folder.
> Probably best to add `ngrok*` to `.gitignore` to avoid pushing them by accident to Github.

done

File: requirements.txt
Comment (hsm207) [reply requirements.txt:5]: yes, we should wait until the PR for twilio is merged.

for now, we can focus on adapting the conversation flows in financial-demo for voice

File: README-voice.md
Comment (hsm207) [reply README-voice.md:13]: I've removed the environment specific commands from the script

File: README-voice.md
Comment (hsm207) [reply README-voice.md:32]: done.

I did not include the steps to steps to get ngrok because I've mentioned it in step 1.

I agree, this PR should be merge only then the Rasa PR is merged and released.

File: README-voice.md
Comment (hsm207) [reply README-voice.md]: There were new constraints added to the supported config parameters.

I've updated the config to resolve the conflict. Feel free to experiment with other values to get better voice experience.

I found that the ones currently set in credentials.yml work well for my setup, which is calling a US Twilio number through Skype.

I would be interested to know your experience.

**Comment by ArjaanBuijk** [comment]
@hsm207 ,
The setup now worked flawlessly, and I was able to talk to it from my cell phone. That was really neat! 

The conversation did not go that smooth, but I think it is ready to be merged, so we then can deploy it and collect the conversations in Rasa Enterprise, and apply some CDD to fix the conversations to work for voice.

Do we have a rasa twilio phone number? Right now I am using my own trial account.

**Comment by mvielkind** [comment]
> The conversation did not go that smooth

@ArjaanBuijk - can you share what about the conversation wasn't smooth? If there are pieces that don't go smoothly we can suggest best practices for how to configure the channel.

**Comment by ArjaanBuijk** [comment]
@mvielkind 
> > The conversation did not go that smooth
> 
> can you share what about the conversation wasn't smooth? If there are pieces that don't go smoothly we can suggest best practices for how to configure the channel.

- The initial greeting is too long for voice. It is a bit annoying.
- I asked about transfer money, and the following happened:
    - It correctly asked who to send it too.
    - I gave the name of a non-existing person
    - It correctly asked again
    - Again I gave the name of a non-existing person
    - It started to explain the slot, but then I got discconnected
    - I called again, and asked about paying a credit card.
        - It said I wasn't done yet with transferring money. So, it was clear that I jumped back into my previous conversation. That is probably not what should happen upon disconnection.

**Comment by hsm207** [comment]
@ArjaanBuijk if the user reconnected before the session expired, shouldn't  jumping back to the previous conversation be the intended behavior?

**Comment by ArjaanBuijk** [comment]
> @ArjaanBuijk if the user reconnected before the session expired, shouldn't jumping back to the previous conversation be the intended behavior?

I am not sure about it. I found it unexpected, because I consider hanging up, or being disconnected, kind of like a restart.

**Pull request closed without merging**
